cmd/dagger/mcp.go

```diff
@@ -55,31 +57,52 @@ func mcpStart(ctx context.Context, engineClient *client.Client) error {
 		return errors.New("currently MCP only works with stdio")
 	}
 	modDef, err := initializeDefaultModule(ctx, engineClient.Dagger())
-	if err != nil {
+	if err != nil && err != errModuleNotFound {
 		return err
 	}
-	q := querybuilder.Query().Client(engineClient.Dagger().GraphQLClient())
-	// TODO: parse user args and pass them to constructor
-	modName := modDef.MainObject.AsObject.Constructor.Name
-	q = q.Root().Select(modName).Select("id")
 
-	var modID string
-	if err := makeRequest(ctx, q, &modID); err != nil {
-		return fmt.Errorf("error instantiating module: %w", err)
+	if err == errModuleNotFound && !core {
+		return fmt.Errorf("%w and --core not specified", errModuleNotFound)
 	}
 
-	q = q.Root().Select("env").Select("with"+modDef.MainObject.AsObject.Name+"Input").
-		Arg("name", modName).
-		Arg("value", modID).
-		Arg("description", "module to expose as an MCP server").
-		Select("id")
+	q := querybuilder.Query().Client(engineClient.Dagger().GraphQLClient())
+	var logMsg string
+	if modDef != nil {
+		// TODO: parse user args and pass them to constructor
+		modName := modDef.MainObject.AsObject.Constructor.Name
+		q = q.Root().Select(modName).Select("id")
+
+		var modID string
+		if err := makeRequest(ctx, q, &modID); err != nil {
+			return fmt.Errorf("error instantiating module: %w", err)
+		}
+
+		q = q.Root().Select("env")
+
+		extraCore := ""
+		if core {
+			q = q.Arg("privileged", core)
+			extraCore = " and Dagger core"
+		}
+
+		q = q.Select("with"+modDef.MainObject.AsObject.Name+"Input").
+			Arg("name", modName).
+			Arg("value", modID).
+			Arg("description", "module to expose as an MCP server").
+			Select("id")
```

Comment:
interesting, it's kinda counterintuitive to me that inputs are what get exposed out through the server